sdk/package/sdk.ts

```diff
@@ -0,0 +1,726 @@
+    const transaction = wasm.Transaction.from_string(txJson);
+    console.log("sdk transaction:" + txJson + ", " + JSON.stringify(transaction));
+
+    // construct asset metadata json by query api
+    const assetIdNumber = uint8ArrayToNumber(asset_id);
+    const asset_meta = await api.query.assetManager.assetIdMetadata(assetIdNumber);
+    // console.log(asset_meta.toHuman());
+    const json = JSON.stringify(asset_meta.toHuman());
+    const jsonObj = JSON.parse(json);
+    console.log("asset metadata:" + json);
+    console.log("asset JSON:" + jsonObj);
+    const decimals = jsonObj["metadata"]["decimals"];
+    const symbol = jsonObj["metadata"]["symbol"];
+    const assetMetadataJson = `{ "decimals": ${decimals}, "symbol": "${PRIVATE_ASSET_PREFIX}${symbol}" }`;
+    console.log("📜asset metadata:" + assetMetadataJson);
+
+    if (onlySign) {
+        const signResult = await sign_transaction(api, wasm, wasmWallet, assetMetadataJson, transaction, network);
+        return signResult;
+    } else {
+        const res = await sign_and_send(api, signer, wasm, wasmWallet, assetMetadataJson, transaction, network);
+        console.log("📜finish private transfer.");
+        return res;
+    }
+}
+
+/// Executes a public transfer from the address of `signer` to the address of `address`,
+/// of the fungible token with AssetId `asset_id`.
+async function public_transfer(api: ApiPromise, signer:string, asset_id:AssetId, address:string, amount:TransferAmount): Promise<any> {
+    try {
+        const asset_id_arr = Array.from(asset_id);
+        const amountBN = new BN(amount).toArray('le', 16);
+        const tx = await api.tx.mantaPay.publicTransfer(
+            { id: asset_id_arr, value: amountBN },
+            address
+        );
+        await tx.signAndSend(signer);
+    } catch (e) {
+        console.log("Failed to execture public transfer.");
+        console.error(e);
+    }
+}
+
+/// Using sign on wallet and using signAndSend to polkadot.js transaction
+/// This version is using `null` asset metdata. Only meaningful for to_private.
+const sign_and_send_without_metadata = async (wasm: any, api: ApiPromise, signer: string, wasmWallet: Wallet, transaction: any, network: Network): Promise<void> => {
+    const signed_transaction = await sign_transaction(api,wasm,wasmWallet,null,transaction,network);
+    for (let i = 0; i < signed_transaction.txs.length; i++) {
+        try {
+            await signed_transaction.txs[i].signAndSend(signer, (status, events) => { });
+        } catch (error) {
+            console.error('Transaction failed', error);
+        }
+    }
+}
+
+/// Signs the a given transaction returning posts, transactions and batches.
+/// assetMetaDataJson is optional, pass in null if transaction should not contain any.
+const sign_transaction = async (api: ApiPromise, wasm: any, wasmWallet: Wallet, assetMetadataJson: any, transaction: Transaction, network: Network) => {
```

Comment:
we use the word `signer` in this file to refer to both manta signer and the polkadot.js api signer, we should try to disambiguate these when naming variables and functions